update_requirements.sh

```diff
@@ -40,25 +40,28 @@ fi
 eval "$(pyenv init --path)"
 
 echo "update pip"
-pip install pip==21.3.1
-pip install pipenv==2021.11.23
+pip install pip==22.3
+pip install poetry==1.2.2
 
+poetry config virtualenvs.prefer-active-python true --local
+
+POETRY_INSTALL_ARGS=""
 if [ -d "./xx" ]; then
-  echo "WARNING: using xx Pipfile ******"
-  export PIPENV_SYSTEM=1
-  export PIPENV_PIPFILE=./xx/Pipfile
+  echo "WARNING: using xx dependency group, installing globally"
+  export POETRY_VIRTUALENVS_CREATE=false
+  POETRY_INSTALL_ARGS="--with xx --sync "
 fi
 
-if [ -z "$PIPENV_SYSTEM" ]; then
+echo "pip packages install..."
+poetry install --no-root $POETRY_INSTALL_ARGS
+
+if [ -z "$PIPENV_SYSTEM" ] && [ -z "$POETRY_VIRTUALENVS_CREATE" ]; then
   echo "PYTHONPATH=${PWD}" > .env
-  RUN="pipenv run"
+  RUN="poetry run"
 else
   RUN=""
 fi
 
-echo "pip packages install..."
-pipenv sync --dev
-pipenv --clear
```

Comment:
does poetry not have an equivalent of this?

Reply:
maybe we can do `pip cache purge` and `poetry cache clear --all PyPI`

Reply:
this makes a big difference in CI, so I'd like that resolved before merging this

Reply:
I just tested it locally but it always prompted me to type `Y/n` even when I passed the `--no-interaction` option... supposedly fixed in Poetry 1.2.1 https://github.com/python-poetry/poetry/releases/tag/1.2.1

Reply:
we already do `rm -rf /root/.cache` in the Dockerfile which should clear this anyway?

Reply:
what about installing with `--no-cache`?